app/controllers/profiles_controller.rb

```diff
@@ -13,13 +13,9 @@ def update
   end
 
   def search
-    @results = []
-    Profile.all.each do |profile|
-      if (@profile.distance_to(profile.postcode) <= params[:distance].to_f) && (profile != @profile)
-        @results << [profile, @profile.distance_to(profile.postcode).round(2)]
-      end
-    end 
-     @results.sort!{|a, b| a[1] <=> b[1]}
+    @results = Profile.near(@profile.postcode, params[:distance].to_f).to_a
```

Comment:
I think that we'd be better off using a `where` clause to find profiles that are not related to `current_customer`. Since these models aren't linked yet I'll merge this but I think you ought to add a bug/chore in pivotal to make a note to fix this up later.